clang/test/CodeGenSYCL/Inputs/sycl.hpp

```diff
@@ -281,6 +281,11 @@ ATTR_SYCL_KERNEL void kernel_single_task(const KernelType &kernelFunc) {
   kernelFunc();
 }
 
+template <typename KernelName = auto_name, typename KernelType>
+ATTR_SYCL_KERNEL void kernel_single_task_2017(KernelType kernelFunc) {
```

Comment:
This was added to differentiate SYCL 2017 kernel invocation function from SYCL 2020.
The SYCL 2020 version supported passing kernel functions by reference, while SYCL 2017 version supported passing kernel function by value.